src/app/launcher/step-indicator/step-indicator.component.html

```diff
@@ -2,8 +2,11 @@
   <input class="f8launcher-application-text-field_input"
          name="applicationTitle" placeholder="New Application" type="text"
          [(ngModel)]="launcherComponent.summary.dependencyCheck.projectName"
-         (keyup.enter)="$event.target.blur();">
+         (keyup.enter)="$event.target.blur();" 
+         pattern="[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+         #check="ngModel" minlength="4">
```

Comment:
The application name is editable in 4 places; here, the summary screen, the import overlay, and the new application overlay. 

Considering this is editable in multiple places, it would be better to create a service to validate the application name. That way, if the validation changes, the code only needs to be updated in one place. Then, we don't need to update the demo or OSIO, just add the service.